fastlane/Fastfile

```diff
@@ -698,21 +698,26 @@ end
   # selected for now, there will be no credentials race condition.
   # -----------------------------------------------------------------------------------
   # Usage:
-  # bundle exec fastlane run_ui_tests [device:<Name of the iOS Simulator >]
+  # bundle exec fastlane run_ui_tests [device:<Name of the iOS Simulator>] [test_account:<Test account email>]
   #
   # Example:
-  # bundle exec fastlane run_ui_tests device:"iPhone 12"
+  # bundle exec fastlane run_ui_tests device:"iPhone 12" test_account:"test.account@test.com"
   #####################################################################################
   desc "Run UI tests"
   lane :run_ui_tests do | options |
-  scan(
-    workspace: workspace_name,
-    scheme: UI_TESTS_SCHEME,
-    device: options[:device] || "iPhone 12",
-    prelaunch_simulator: true,
-    reset_simulator: true,
-    only_testing: "SimplenoteUITests/SimplenoteUISmokeTestsLogin"
-  )
+    begin
+      scan(
+        workspace: workspace_name,
+        scheme: UI_TESTS_SCHEME,
+        device: options[:device] || "iPhone 12",
+        prelaunch_simulator: true,
+        reset_simulator: true,
+        only_testing: "SimplenoteUITests/SimplenoteUISmokeTestsTrash",
+        xcargs: "UI_TEST_ACCOUNT='#{options[:test_account] || ""}'"
+      )
+    rescue => ex
+       UI.error(ex)
+    end
```

Comment:
Update: while testing the `fastlane` invocation, I passed a wrong email on purpose. This code did what it was told to do, that is, captured and reported the error, but did not fail the lane.

![image](https://user-images.githubusercontent.com/1218433/117093962-c7e7d300-ada5-11eb-969f-8c015b0dbe89.png)

From my point of view, this is not a desirable behavior because, if tests fail, we want to know about it. Am I missing some context?

Thanks 😄

Reply:
Thank you @mokagio ! Yes, there is really a reason behind this: in my next PR there will be another lane, that picks the free account, and calls `run_ui_tests` lane and passes the user email to it. Regardless of tests execution status, I need to make sure that the test account is always freed at the end. Without `begin - rescue` block, if the there's a fail, the execution will not cruise back to the `pick_test_account_and_run_ui_tests` lane in order to mark the account as free.

This might be not the only way to achieve this, but the one that I found working. If this goes against the best practices, I can remove this, and we can discuss possible options in the next PR?